commands/command_checkout.go

```diff
@@ -75,6 +94,61 @@ func checkoutCommand(cmd *cobra.Command, args []string) {
 	singleCheckout.Close()
 }
 
+func checkoutConflict(file string, stage int) {
+	singleCheckout := newSingleCheckout(cfg.Git, "")
+	if singleCheckout.Skip() {
+		fmt.Println("Cannot checkout LFS objects, Git LFS is not installed.")
+		return
+	}
+
+	ref, err := git.ResolveRef(fmt.Sprintf(":%d:%s", stage, file))
+	if err != nil {
+		Exit("Could not checkout (are you not in the middle of a merge?): %v", err)
+	}
+
+	scanner, err := git.NewObjectScanner()
+	if err != nil {
+		Exit("Could not create object scanner: %v", err)
+	}
+
+	if !scanner.Scan(ref.Sha) {
+		Exit("Could not find object %q", ref.Sha)
+	}
+
+	ptr, err := lfs.DecodePointer(scanner.Contents())
+	if err != nil {
+		Exit("Could not find decoder pointer for object %q: %v", ref.Sha, err)
+	}
+
+	p := &lfs.WrappedPointer{Name: file, Pointer: ptr}
+
+	singleCheckout.RunToPath(p, checkoutTo)
+	singleCheckout.Close()
+}
+
+func whichCheckout() (stage int, err error) {
```

Comment:
What do you think about returning a named version of `int`, perhaps like:

```go
type CheckoutStage int

const (
        CheckoutStageUnknown CheckoutStage = iota
        CheckoutStageBase
        CheckoutStageOurs
        CheckoutStageTheirs
)
```

Potentially this could type definition could also go in package `git`. It looks like there is a facility from `cobra` that provides us a way to do this sort of "assign a value to this location in memory based not on its type" via [`func Var()`](https://godoc.org/github.com/spf13/pflag#Var). I think we'd have to definite an implementation of `type Value interface`, but that seems doable.

Reply:
Actually, I don't think `Var` does what we want.  I remember looking at it, and now that I'm looking again, it seems to require an argument, which we aren't passing in this case.  What we really need is a flag type that sets a variable to have a specific value, which I don't think `cobra` has.